docs/content/docs/data-structures/counter.md

```diff
@@ -1,5 +1,196 @@
+
+Let's say that two users are collaborating on an app with a counter widget.
+We will refer to these users as User A and User B.
+The current value of that widget is 42.
+
+What if User A and User B both simultaneously press the `+` button in their UI to increment the current value by 1?
+Behind the scenes, this increment is implemented by writing the updated value to the map.
+Each edit is then sequenced and broadcast to other collaborators.
+
+For the purpose of this example, we are ignoring the fact that two operations will never really occur at exactly the same time.
+This is intentional.
+Since increments and decrements are [commutative](https://en.wikipedia.org/wiki/Commutative_property), it shouldn't matter in what order two nearly concurrent increment operations occur; the result _should_ be the same.
+
+Here, we would expect that, after both users have processed all incoming edits, the resulting value of the counter would be 44.
+Each user pressed the `+` button once.
+42 + 1 + 1 = **44**, right?
+Again, note that we are ignoring which `+ 1` came from which user, as the order should not matter.
+
+Unfortunately, because `SharedMap` employs a _last-write-wins_ merge strategy, if User A and User B make their edits at the same time, one of the two updates will be sequnced after the other, and that value will win.
+Regardless of which user's edit was sequenced first and which was sequenced last, each user's client will update their map entry to be 42 (the current value they see) + 1 = **43**, and the latter of the two operations to be sequenced will clobber the first.
+
+So in this case, both users would see the value 43, rather than 44.
+
+This is a problem!
+If this shared counter value was being used to track store inventory, for example, that inventory would now be incorrect!
+
+The solution to this problem is a specialized DDS that tracks changes to the shared value as _increments_ and _decrements_, which can be summed together in any order and still reach eventual consistency.
+
+## Creation
+
+The `FluidContainer` provides a container schema for defining which DDSes you would like to load from it.
+It provides two separate fields for establishing an initial roster of objects and dynamically creating new ones.
+
+- For general guidance on using the `ContainerSchema`, see [Data modeling]({{< relref "data-modeling.md" >}}).
+- For guidance on how to create and load a Fluid container using a service-specific client, see [Containers - Creating and loading]({{< relref "containers.md#creating--loading" >}}).
+
+The following example illustrates how you can use the `ContainerSchema` with `SharedCounter`.
+
+There are two key workflows to consider here.
+
+1. Creating a new Fluid Container.
+2. Connecting to an existing Fluid Container.
+
+In either case, you will first define the schema.
+
+```javascript
+const schema = {
+    initialObjects: {
+        sharedCounter: SharedCounter,
+    }
+}
```

Comment:
Maybe worth calling out dynamic usage of DDSes in our examples.

Reply:
you have it below, nevermind. Maybe worth combining the two, so the user does not think you either have to use one or the other. Or as I thought initially - if I added factory in `initialObjects`, I did not need to add it to dynamic list. No strong preferences here, up to you.

Reply:
Ah, good call. I structured the flow here to mirror the `map` article, so I'd prefer not to change that (unless we want to change both articles). But I will update the dynamic code block to still include the initial objects part.